nestjs/src/gratitudes/gratitudes.service.ts

```diff
@@ -26,19 +26,20 @@ export class GratitudesService {
       throw new BadRequestException('Badge not allowed');
     }
 
-    const { id } = await this.repository.save({
-      fromUserId: authUser.id,
-      toUserId: data.userId,
-      comment: data.comment,
-      badgeId: data.badgeId,
-      courseId: data.courseId,
-    });
-    const result = await this.repository.findOneOrFail(id, {
-      relations: ['fromUser', 'toUser', 'course', 'course.discordServer'],
-    });
+    await Promise.all(
+      data.userIds.map(async toUserId => {
```

Comment:
could you please move it to a separate function for readability?

I mean create a function that creates feedback and send notification for a single user